src/librustc_middle/mir/interpret/int.rs

```diff
@@ -0,0 +1,36 @@
+use rustc_target::abi::Size;
+
+#[derive(Copy, Clone)]
+/// A type for representing any integer. Only used for printing.
+// FIXME: Use this for the integer-tree representation needed for type level ints and
+// const generics?
+pub struct ConstInt {
```

Comment:
In terms of organization, maybe it would make sense to have a `const` file here and move both `ConstValue` and `ConstInt` there, separating the Miri-engine-internal types from the CTFE-external types?

Reply:
I moved it to the `ty` module